src/title.js

```diff
@@ -0,0 +1,441 @@
+	 */
+	setBody( data ) {
+		const editor = this.editor;
+		const root = editor.model.document.getRoot();
+		const range = editor.model.createRange(
+			editor.model.createPositionAt( root.getChild( 0 ), 'after' ),
+			editor.model.createPositionAt( root, 'end' )
+		);
+
+		editor.model.insertContent( editor.data.parse( data ), range );
+	}
+
+	/**
+	 * Returns the body of the document.
+	 *
+	 * @returns {String} Body of the document.
+	 */
+	getBody() {
+		const root = this.editor.model.document.getRoot();
+		const viewWriter = new ViewWriter();
+
+		// model -> view
+		const viewDocumentFragment = this.editor.data.toView( root );
+
+		// Remove title.
+		viewWriter.remove( viewWriter.createRangeOn( viewDocumentFragment.getChild( 0 ) ) );
+
+		// view -> data
+		return this.editor.data.processor.toData( viewDocumentFragment );
+	}
+
+	/**
+	 * Returns `title` element when is in the document. Returns `undefined` otherwise.
+	 *
+	 * @private
+	 * @returns {module:engine/model/element~Element|undefined}
+	 */
+	_getTitleElement() {
+		const root = this.editor.model.document.getRoot();
+
+		for ( const child of root.getChildren() ) {
+			if ( child.is( 'title' ) ) {
+				return child;
+			}
+		}
+	}
+
+	/**
+	 * Model post-fixer callback that ensures `title` has only one `title-content` child.
+	 * All additional children should be moved after `title` element and renamed to a paragraph.
+	 *
+	 * @private
+	 * @param {module:engine/model/writer~Writer} writer
+	 * @returns {Boolean}
+	 */
+	_fixTitleContent( writer ) {
+		const title = this._getTitleElement();
+
+		if ( !title ) {
+			return false;
+		}
+
+		const titleChildren = Array.from( title.getChildren() );
+		let hasChanged = false;
+
+		// Skip first child because it is an allowed element.
+		titleChildren.shift();
+
+		for ( const titleChild of titleChildren ) {
+			writer.move( writer.createRangeOn( titleChild ), title, 'after' );
+			writer.rename( titleChild, 'paragraph' );
+			hasChanged = true;
+		}
+
+		return hasChanged;
+	}
+
+	/**
+	 * Model post-fixer callback that takes care about creating and keeping `title` element as a first child in a root.
+	 *
+	 * @private
+	 * @param {module:engine/model/writer~Writer} writer
+	 * @returns {Boolean}
+	 */
+	_fixTitleElement( writer ) {
+		const model = this.editor.model;
+		const modelRoot = model.document.getRoot();
+		let hasChanged = false;
+		let numberOfMovedElements = 0;
+
+		// Loop through root children and take care about a proper position of a title element.
+		// Title always has to be the first element in the root.
+		for ( const rootChild of Array.from( modelRoot.getChildren() ) ) {
+			// If the first element is not a title we need to fix it.
+			if ( rootChild.index === 0 && !rootChild.is( 'title' ) ) {
+				const title = this._getTitleElement();
+
+				// Change first element to the title if it can be a title.
+				if ( Title.titleLikeElements.has( rootChild.name ) ) {
```

Comment:
I might be missing some requirements here but I thins that existing title should be used as a title (I've added some tests) for this case. So they might be invalid but I feel that if we have a `<title>` in the body we might like to have it as a first element anyway. OTOH it may be better to leave the content as is (visually) and fix the tests expects only.